cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/ForEachStatement.kt

```diff
@@ -25,22 +25,63 @@
  */
 package de.fraunhofer.aisec.cpg.graph.statements
 
-import de.fraunhofer.aisec.cpg.graph.AST
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.edge.PropertyEdge
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.DeclaredReferenceExpression
 import java.util.Objects
 
-class ForEachStatement : Statement() {
+class ForEachStatement : Statement(), BranchingNode, StatementHolder {
     /**
      * This field contains the iteration variable of the loop. It can be either a new variable
      * declaration or a reference to an existing variable.
      */
-    @AST var variable: Statement? = null
+    @AST
+    var variable: Statement? = null
+        set(value) {
+            if (value is DeclaredReferenceExpression) {
+                value.access = AccessValues.WRITE
+            }
+            field = value
+        }
 
     /** This field contains the iteration subject of the loop. */
     @AST var iterable: Statement? = null
 
     /** This field contains the body of the loop. */
     @AST var statement: Statement? = null
 
+    override val branchingDecision: Node?
+        get() = iterable
+
+    override var statementEdges: MutableList<PropertyEdge<Statement>>
```

Comment:
What is the reason for implementing statement holder here?

Reply:
I am fine with the term. How would be a `branchedBy`?

Reply:
I think I needed a `StatementHolder` for the Fluent-DSL which I use in a test. But maybe I just don't understand when to use it.

Reply:
I tried to remove the `StatementHolder` and rework the fluent DSL part for for-each a little bit. It's not really pretty but creating the node works. However, now the `ControlDependenceGraphPassTest` seems to hang; which is weird because the for-each statement that is created is correct. Maybe it was not correct before and should have never worked? :D

Reply:
Removing the `StatementHolder` adds the nodes in the parent and the for-each statement which means that it's necessary right now. However, I managed to fix the test even in the presence of the bug/loop.